docker/Dockerfile

```diff
@@ -1,20 +1,56 @@
-FROM golang:1.13.8 AS dependencies
+# Copyright (c) 2020 Red Hat, Inc.
+# This program and the accompanying materials are made
+# available under the terms of the Eclipse Public License 2.0
+# which is available at https://www.eclipse.org/legal/epl-2.0/
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Contributors:
+#   Red Hat, Inc. - initial API and implementation
+#
 
-WORKDIR /kubernetes-image-puller
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8/go-toolset
+FROM registry.access.redhat.com/ubi8/go-toolset:1.13.4 as builder
+ENV PATH=/opt/rh/go-toolset-1.13/root/usr/bin:$PATH \
+    GO111MODULE=on
+
+ARG BOOTSTRAP=true
+ENV BOOTSTRAP=${BOOTSTRAP}
+
+
+USER root
 
-# Populate the module cache based on the go.{mod,sum} files.
+WORKDIR /kubernetes-image-puller
 COPY go.mod .
 COPY go.sum .
-RUN go mod download
-
-FROM dependencies AS builder
+# built in Brew, use tarball in lookaside cache; built locally, comment this out
+# COPY resources.tgz /tmp/resources.tgz
+# build locally, fetch mods
+RUN if [[ ${BOOTSTRAP} != "false" ]]; then \
+      go mod download; \
+    elif [[ -f /tmp/resources.tgz ]]; then \
+      tar xvf /tmp/resources.tgz -C /; \
+      rm -f /tmp/resources.tgz; \
+    fi
 
 COPY . .
-# RUN go test -v ./cfg... ./pkg... ./utils... \
-#     && GOOS=linux go build -v -o ./bin/${BINARY_NAME} ./cmd/main.go
-RUN make build
 
+RUN adduser appuser && \
+    make build 
 
-FROM gcr.io/distroless/base
-COPY --from=builder "/kubernetes-image-puller/bin/kubernetes-image-puller" "/"
+# https://access.redhat.com/containers/?tab=tags#/registry.access.redhat.com/ubi8-minimal
+FROM registry.access.redhat.com/ubi8-minimal:8.2-267
+USER root
+# CRW-528 copy actual cert
+COPY --from=builder /etc/pki/ca-trust/extracted/pem/tls-ca-bundle.pem /etc/pki/ca-trust/extracted/pem/
+# CRW-528 copy symlink to the above cert
+COPY --from=builder /etc/pki/tls/certs/ca-bundle.crt                  /etc/pki/tls/certs/
+COPY --from=builder /etc/passwd /etc/passwd
+
+USER appuser
```

Comment:
Is this step necessary? Once running on a cluster, the uid won't be `appuser` and `/etc/passwd` won't contain any info about the current user.

It seems to me this could be done with a simple `USER 1001` and potentially a chown on the COPY below if necessary.

Reply:
@nickboldt any comments on that ^

Reply:
afaict as long as the user is valid it doesn't matter what its name/number is. Just make sure that user 1001 exists and any refs to its home dir exist, permissions are set correctly, etc. 

In which other rhel-based dockerfiles do we use `USER 1001` ?

Reply:
My issue is moreso that once this gets deployed to OpenShift, the entrypoint for the container will no longer run as `appuser` or `USER 1001` -- it'll be assigned some random UID. This section is more a no-op than incorrect, I think (i.e. you still won't have a home dir, permissions, etc.). 

I'm fine with it as-is, just pointing out that this could be misleading down the line. If this is included to sync with downstream for simplicity, that obviously outweighs my minor complaint :slightly_smiling_face: 

The whole `USER 1001` thing I started doing because I remember reading some doc that suggested it as a way of asserting that your container does not require privileged access.

Reply:
@ibuziuk if you want to follow Angel's advice here, I'm happy to have the same in downstream. I've never actually *used* this container so I can't speak to what the change would do (if anything), but I suspect the effect would be "less code that does nothing", so +1 to making things better.

Reply:
@nickboldt well, that is the question which team is supposed to maintain rhel based images - for this PR I just used the rhel image that was contributed initially